lib/taskana-core-test/src/test/java/acceptance/task/update/UpdateTaskAccTest.java

```diff
@@ -0,0 +1,582 @@
+            .buildAndStore(taskService);
+    task.setPrimaryObjRef(createObjectReference("COMPANY_A", null, null, "VNR", "7654321"));
+
+    Task updatedTask = taskService.updateTask(task);
+    updatedTask = taskService.getTask(updatedTask.getId());
+
+    assertThat(updatedTask).isNotNull();
+    assertThat(updatedTask.getPrimaryObjRef().getSystem()).isNull();
+    assertThat(updatedTask.getPrimaryObjRef().getSystemInstance()).isNull();
+    assertThat(updatedTask.getPrimaryObjRef().getCompany()).isEqualTo("COMPANY_A");
+    assertThat(updatedTask.getPrimaryObjRef().getType()).isEqualTo("VNR");
+    assertThat(updatedTask.getPrimaryObjRef().getValue()).isEqualTo("7654321");
+  }
+
+  @WithAccessId(user = "user-1-2")
+  @TestFactory
+  Stream<DynamicTest>
+      should_ThrowException_When_MandatoryPrimaryObjectReferenceIsNotSetOrIncomplete()
+          throws Exception {
+    List<Pair<String, ObjectReference>> list =
+        List.of(
+            Pair.of("When Primary Object Reference is null", null),
+            Pair.of(
+                "When Value of Primary Object Reference is null",
+                createObjectReference("COMPANY_A", "SYSTEM_A", "INSTANCE_A", "VNR", null)),
+            Pair.of(
+                "When Type of Primary Object Reference is null",
+                createObjectReference("COMPANY_A", "SYSTEM_A", "INSTANCE_A", null, "1234567")),
+            Pair.of(
+                "When Company of Primary Object Reference is null",
+                createObjectReference(null, "SYSTEM_A", "INSTANCE_A", "VNR", "1234567")));
+
+    ThrowingConsumer<Pair<String, ObjectReference>> test =
+        pair -> {
+          Task task =
+              TaskBuilder.newTask()
+                  .classificationSummary(defaultClassificationSummary)
+                  .workbasketSummary(defaultWorkbasketSummary)
+                  .primaryObjRef(defaultObjectReference)
+                  .buildAndStore(taskService);
+          task.setPrimaryObjRef(pair.getRight());
+          assertThatThrownBy(() -> taskService.updateTask(task))
+              .isInstanceOf(InvalidArgumentException.class);
+        };
+    return DynamicTest.stream(list.iterator(), Pair::getLeft, test);
+  }
+
+  @WithAccessId(user = "user-taskrouter")
+  @Test
+  void should_ThrowException_When_UserIsNotAuthorized() throws Exception {
+    Task task =
+        TaskBuilder.newTask()
+            .classificationSummary(defaultClassificationSummary)
+            .workbasketSummary(defaultWorkbasketSummary)
+            .primaryObjRef(defaultObjectReference)
+            .buildAndStore(taskService, "admin");
+    task.setPrimaryObjRef(
+        taskService.newObjectReference("company", "system", "instance", "value", "type"));
+    ClassificationSummary classificationSummary =
+        defaultTestClassification().buildAndStoreAsSummary(classificationService, "admin");
+    task.setClassificationKey(classificationSummary.getKey());
```

Comment:
You don't need to change two attributes of Task for this test. What do you think about updating something more simple instead? Like for example the note.